### PR TITLE
Bug 1182509 - [Contacts][NGA] Open contacts_matching with window.open

### PR DESCRIPTION
**Pull request by jpruden92**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cdd320306f5a696307308763ade8e0b24e877c1d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fd9a98f0d81325e941afd79e3cd39aed5839a004

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=768496ccf0e7f4e997a140c691e09ed2709ecf75

File: apps/communications/contacts/js/match_service.js [hunk 1]
```diff
@@ -47,42 +54,32 @@
...
     window.dispatchEvent(new CustomEvent('image-loader-resume'));
-    extensionFrame.src = currentURI = null;
   }
 
   window.addEventListener('message', messageHandler);
 
   exports.MatchService = matchService;
 
-}(window));
+}(window));
```
Comment (borjasalguero) [inline apps/communications/contacts/js/match_service.js]: Add one more line!

File: apps/communications/contacts/js/match_service.js [hunk 2]
```diff
@@ -1,39 +1,46 @@
...
+    init();
   };
 
-  function close(messageId, additionalMessageId) {
-    extensionFrame.addEventListener('transitionend', function tclose() {
-      extensionFrame.removeEventListener('transitionend', tclose);
-      extensionFrame.classList.add('hidden');
-      unload();
+  function init(){
+    extensionFrame.addEventListener('load', function(){
```
Comment (borjasalguero) [inline apps/communications/contacts/js/match_service.js]: Remove event listener after this line.

File: apps/communications/contacts/js/match_service.js [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
+      extensionFrame.postMessage({
+        type: 'sync'
+      }, CONTACTS_APP_ORIGIN);
+      extensionFrame.onunload = function(){
+        window.postMessage({
```
Comment (borjasalguero) [inline apps/communications/contacts/js/match_service.js]: Clean `onunload` after this.

File: apps/communications/contacts/js/views/form.js [hunk 4]
```diff
@@ -872,6 +872,7 @@ contacts.Form = (function() {
 
             case 'merge_duplicate_contacts':
               window.removeEventListener('message', mergeHandler);
+              Contacts.goBack();
```
Comment (borjasalguero) [inline apps/communications/contacts/js/views/form.js]: This is good by now due to this entire JS is going to dissappear :).

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ff2bc94445278597ebbacd873442d4455a608216

File: apps/communications/contacts/js/match_service.js [hunk 5]
```diff
@@ -1,39 +1,48 @@
...
+      extensionFrame.removeEventListener('load', fn);
+      extensionFrame.postMessage({
+        type: 'sync'
+      }, CONTACTS_APP_ORIGIN);
+      extensionFrame.onunload = function(){
+        window.postMessage({
+          type: 'window_close'
+        }, CONTACTS_APP_ORIGIN);
+      };
+      extensionFrame.onunload = null;
```
Comment (borjasalguero) [inline apps/communications/contacts/js/match_service.js]: This must be inside the `onunload` method.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c41b6f4d2507baa38564d9b653156363a43d291

File: apps/communications/contacts/js/match_service.js [hunk 6]
```diff
@@ -1,39 +1,48 @@
...
+
   var currentURI;
 
   matchService.match = function(contactId) {
-    extensionFrame.src = currentURI = 
-      '/contacts/views/matching/matching_contacts.html?contactId=' + contactId;
+    extensionFrame = window.open(
+      '/contacts/views/matching/matching_contacts.html?contactId=' +
+      contactId, 'Contacts match');
+    currentURI = '/contacts/views/matching/matching_contacts.html?contactId=' +
```
Comment (arcturus) [inline apps/communications/contacts/js/match_service.js]: make currentURI variable available before you do the window.open, and use that variable in the window.open

File: apps/communications/contacts/js/match_service.js [hunk 7]
```diff
@@ -1,39 +1,48 @@
...
 (function(exports) {
   var matchService = {};
 
   const CONTACTS_APP_ORIGIN = location.origin;
-
-  var extensionFrame = document.createElement('iframe');
-  extensionFrame.id = 'match-extension';
-  document.body.appendChild(extensionFrame);
   
+  var extensionFrame = null;
```
Comment (arcturus) [inline apps/communications/contacts/js/match_service.js]: Should we still call it `extensionFrame`? Now we could use name that suits better the situation

File: apps/communications/contacts/js/match_service.js [hunk 8]
```diff
@@ -1,39 +1,48 @@
...
+      contactId, 'Contacts match');
+    currentURI = '/contacts/views/matching/matching_contacts.html?contactId=' +
+      contactId;
+    init();
   };
 
   matchService.showDuplicateContacts = function() {
-    extensionFrame.src = currentURI = 
-      '/contacts/views/matching/matching_contacts.html';
+    extensionFrame = window.open(
```
Comment (arcturus) [inline apps/communications/contacts/js/match_service.js]: are the showDuplicateContacts and match methods the same?

**Pull request closed without merging**